src/services/exchange/exchange.service.ts

```diff
@@ -41,14 +44,29 @@ export class ExchangeService {
     return toExchangeRate / fromExchangeRate;
   }
 
-  private async getExchangeResult(): Promise<ExchangeResult> {
+  async getFiatCodes(): Promise<string[]> {
+    const fiatCodesResult: FiatCodesExchangeResult =
+      (await this.getExchangeResult('symbols')) as FiatCodesExchangeResult;
+
+    return Object.keys(fiatCodesResult.symbols);
+  }
+
+  private async getExchangeResult(
```

Comment:
I've modified this function to serve both endpoints and avoid repeating ourselves. But in fact I think we could extract this function and create something like a `ExchangeClient` to encapsulate the data retrieval and make `ExchangeService` agnostic. Wdyt?

Reply:
I'd separate both calls. We can do this in the `ExchangeService` for now without creating another class (at least for now).

But I'd split both calls as we might want to handle error cases differently (and I also think that mapping 1 endpoint to 1 function makes it easier to read overall).

Reply:
In order to not read from the Config everytime we call these functions I'd suggest the following:

```typescript
  private readonly baseUrl: string;
  private readonly apiKey: string;

  constructor(
    @Inject(IConfigurationService)
    private readonly configurationService: IConfigurationService,
    @Inject(NetworkService) private readonly networkService: INetworkService,
    private readonly httpErrorHandler: HttpErrorHandler,
  ) {
    this.baseUrl =
      this.configurationService.getOrThrow<string>('exchange.baseUri');
    this.apiKey =
      this.configurationService.getOrThrow<string>('exchange.apiKey');
  }
```